spec/rubocop/cop/rails/reflection_class_name_spec.rb

```diff
@@ -124,6 +124,12 @@
     RUBY
   end
 
+  it 'does not register an offense when parameter value is a method call on an object in a variable' do
+    expect_no_offenses(<<~RUBY)
+      has_many :accounts, class_name: the_class.name
```

Comment:
The ability to call the `name` method suggests that, similar to `has_many :accounts, class_name: Account.name`, an offense should be registered:
https://github.com/rubocop/rubocop-rails/pull/1180/files#diff-9d8842c039d0144f6548e139d525048bb9e0651ba63906ca875e35ff52140498R16-R25

Reply:
I understand the purpose of this cop to be to avoid the use of constant references, which lead to loading all the referenced classes.

`Account.name` will resolve `Account`, `some_class_in_a_variable.name` will not load the class (it will already have been loaded).

I can see how this is kind of an edge case though since it does seem to require the loading of a class (unless the `name` method here is not `Class#name`).

I wonder - isn't your concern covered by `it 'registers an offense when parameter value is a local variable assigned a constant'`? 

I can look into this, change this test case to use different variable / method names and add another test case for

```ruby
expect_offense(<<~RUBY)
  the_class = Account

  has_many :accounts, class_name: the_class.name
                                      ^^^^^^^^^^^^^^^^^^^^^^ Use a string value for `class_name`.
RUBY
```

(Although I think this makes me think that the message shouldn't be "Use a string value for `class_name`", but rather "Don't use a constant reference for `class_name`")

Reply:
OK, I tried it -

```ruby
  the_class = Account

  has_many :accounts, class_name: the_class.name
```

does not register an offense even on main. So for now I've just changed the variable and method name in this test. I believe the behavior of the cop is as expected except for the above example which didn't work before either. But it's better to have a false negative than a false positive.